scripts/ensure-correct-devtools-protocol-package.ts

```diff
@@ -0,0 +1,83 @@
+/**
+ * Copyright 2020 Google Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * This script ensures that the pinned version of devtools-protocol in
+ * package.json is the right version for the current revision of Chromium that
+ * Puppeteer ships with.
+ *
+ * The devtools-protocol package publisher runs every hour and checks if there
+ * are protocol changes. If there are, it will be versioned with the revision
+ * number of the commit that last changed the .pdl files.
+ *
+ * Chromium branches/releases are figured out at a later point in time, so it's
+ * not true that each Chromium revision will have an exact matching revision
+ * version of devtools-protocol. To ensure we're using a devtools-protocol that
+ * is aligned with our revision, we want to find the largest package number
+ * that's <= the revision that Puppeteer is using.
+ *
+ * This script uses npm's `view` function to list all versions in a range and
+ * find the one closest to our Chromium revision.
+ */
+
+import { PUPPETEER_REVISIONS } from '../src/revisions';
+import { execSync } from 'child_process';
+
+import packageJson from '../package.json';
+
+const currentProtocolPackageInstalledVersion =
+  packageJson.dependencies['devtools-protocol'];
+
+/**
+ * Ensure that the devtools-protocol version is pinned.
+ */
+if (!currentProtocolPackageInstalledVersion[0].match(/[0-9]/)) {
```

Comment:
Nit: we don't need the match result, so let's use `RegExp#test`

```suggestion
if (/^[^0-9]/.test(currentProtocolPackageInstalledVersion) {
```

(`[^0-9]` could be `\D` but I actually like spelling it out more explicitly)

Reply:
ah nice